Re-enable wasm32-unknown-emscripten target.

Alexhuszagh opened the pull request:
Uses a trick described in rust-lang/rust/issues#85821#issuecomment-969369677. Note that this still allows it to run, despite there seemingly being no entry point: this has no effect on the resulting code.

Related to #609. Partially fixes #839.

Alexhuszagh commented:
Note that `asmjs-unknown-emscripten` cannot be re-enabled, due to issues described in the updated CI action and in #839. I've tested it against `stdexcept` in a C++ target, which seemingly was the issue before, and it works fine. The file compiles, but when run properly complains that Rust cannot catch foreign exceptions:

```cpp
#include <iostream>
#include <stdexcept>

extern "C" {
  void hello() {
    std::cout << "Hello, world!" << std::endl;
     throw std::invalid_argument( "received negative value" );
  }
}
```

Alexhuszagh commented:
bors try --target wasm32-unknown-emscripten

Emilgardis commented on CHANGELOG.md:
```diff
@@ -31,6 +31,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 ### Changed
 
+- #840 - re-enabled `wasm32-unknown-emscripten` target.
```
remove, this is not a change from 0.2.1

Emilgardis reviewed (approved):
should be added to readme too, but otherwise, nice catch!

Alexhuszagh commented:
I've updated the README and also updated the extract target info for this, since I also forgot to do this for solaris. Feel free to merge this whenever ready.

Emilgardis commented:
bors r+

Jules-Bertholet commented:
The upstream Rust issue has since been resolved (https://github.com/rust-lang/rust/issues/8582#issuecomment-969369677), and Emscripten has also made a change to support older Rust versions again (https://github.com/emscripten-core/emscripten/pull/17222). The emscripten fix has made it to the latest release (https://hub.docker.com/layers/emsdk/emscripten/emsdk/3.1.14/images/sha256-4c3e0a0dac61430b719e82118ae9b2c7480902a2713267e80fa296d39f7ab921). So this PR should probably be reverted in favor of a bump to the emscripten version used in the Dockerfile
